File: src/java/util/stream/Stream.java
public abstract interface class java.util.stream.Stream
{
    // Constructors
    // Methods
    public static java.util.stream.Stream$Builder builder();
    public static java.util.stream.Stream concat(java.util.stream.Stream, java.util.stream.Stream);
    public static java.util.stream.Stream empty();
    public static java.util.stream.Stream generate(java.util.function.Supplier);
    public static java.util.stream.Stream iterate(java.lang.Object, java.util.function.UnaryOperator);
    public static java.util.stream.Stream of(java.lang.Object);
    public static transient java.util.stream.Stream of([Ljava.lang.Object;);
    public abstract boolean allMatch(java.util.function.Predicate);
    public abstract boolean anyMatch(java.util.function.Predicate);
    public abstract java.lang.Object collect(java.util.function.Supplier, java.util.function.BiConsumer, java.util.function.BiConsumer);
    public abstract java.lang.Object collect(java.util.stream.Collector);
    public abstract long count();
    public abstract java.util.stream.Stream distinct();
    public abstract java.util.stream.Stream filter(java.util.function.Predicate);
    public abstract java.util.Optional findAny();
    public abstract java.util.Optional findFirst();
    public abstract java.util.stream.Stream flatMap(java.util.function.Function);
    public abstract java.util.stream.DoubleStream flatMapToDouble(java.util.function.Function);
    public abstract java.util.stream.IntStream flatMapToInt(java.util.function.Function);
    public abstract java.util.stream.LongStream flatMapToLong(java.util.function.Function);
    public abstract void forEach(java.util.function.Consumer);
    public abstract void forEachOrdered(java.util.function.Consumer);
    public abstract java.util.stream.Stream limit(long);
    public abstract java.util.stream.Stream map(java.util.function.Function);
    public abstract java.util.stream.DoubleStream mapToDouble(java.util.function.ToDoubleFunction);
    public abstract java.util.stream.IntStream mapToInt(java.util.function.ToIntFunction);
    public abstract java.util.stream.LongStream mapToLong(java.util.function.ToLongFunction);
    public abstract java.util.Optional max(java.util.Comparator);
    public abstract java.util.Optional min(java.util.Comparator);
    public abstract boolean noneMatch(java.util.function.Predicate);
    public abstract java.util.stream.Stream peek(java.util.function.Consumer);
    public abstract java.lang.Object reduce(java.lang.Object, java.util.function.BiFunction, java.util.function.BinaryOperator);
    public abstract java.lang.Object reduce(java.lang.Object, java.util.function.BinaryOperator);
    public abstract java.util.Optional reduce(java.util.function.BinaryOperator);
    public abstract java.util.stream.Stream skip(long);
    public abstract java.util.stream.Stream sorted();
    public abstract java.util.stream.Stream sorted(java.util.Comparator);
    public abstract [Ljava.lang.Object; toArray();
    public abstract [Ljava.lang.Object; toArray(java.util.function.IntFunction);
    // Fields
}

File: src/java/util/stream/Stream.ts
export abstract class Stream{
// Methods
public static builder(): Stream$Builder {
return Packages.java.util.stream.Stream.builder()
}
public static concat(str: Stream, str_1: Stream): Stream {
return Packages.java.util.stream.Stream.concat(str, str_1)
}
public static empty(): Stream {
return Packages.java.util.stream.Stream.empty()
}
public static generate(sup: Supplier): Stream {
return Packages.java.util.stream.Stream.generate(sup)
}
public static iterate(obj: Object, una: UnaryOperator): Stream {
return Packages.java.util.stream.Stream.iterate(obj, una)
}
public static of(obj: Object): Stream {
return Packages.java.util.stream.Stream.of(obj)
}
public static transient of(obj: Object[]): Stream {
return Packages.java.util.stream.Stream.of(obj)
}
public abstract allMatch(pre: Predicate): boolean {}
public abstract anyMatch(pre: Predicate): boolean {}
public abstract collect(sup: Supplier, bic: BiConsumer, bic_1: BiConsumer): Object {}
public abstract collect(col: Collector): Object {}
public abstract count(): number {}
public abstract distinct(): Stream {}
public abstract filter(pre: Predicate): Stream {}
public abstract findAny(): Optional {}
public abstract findFirst(): Optional {}
public abstract flatMap(fun: Function): Stream {}
public abstract flatMapToDouble(fun: Function): DoubleStream {}
public abstract flatMapToInt(fun: Function): IntStream {}
public abstract flatMapToLong(fun: Function): LongStream {}
public abstract forEach(con: Consumer): void {
this.forEach(con)
}
public abstract forEachOrdered(con: Consumer): void {
this.forEachOrdered(con)
}
public abstract limit(lon: number): Stream {}
public abstract map(fun: Function): Stream {}
public abstract mapToDouble(tod: ToDoubleFunction): DoubleStream {}
public abstract mapToInt(toi: ToIntFunction): IntStream {}
public abstract mapToLong(tol: ToLongFunction): LongStream {}
public abstract max(com: Comparator): Optional {}
public abstract min(com: Comparator): Optional {}
public abstract noneMatch(pre: Predicate): boolean {}
public abstract peek(con: Consumer): Stream {}
public abstract reduce(obj: Object, bif: BiFunction, bin: BinaryOperator): Object {}
public abstract reduce(obj: Object, bin: BinaryOperator): Object {}
public abstract reduce(bin: BinaryOperator): Optional {}
public abstract skip(lon: number): Stream {}
public abstract sorted(): Stream {}
public abstract sorted(com: Comparator): Stream {}
public abstract toArray(): Object[] {}
public abstract toArray(int: IntFunction): Object[] {}
}